### ISSUE TITLE
Mouse clicks aren't registering (due to drag?)

### ISSUE THREAD

**Issue by skerit** [labwc/labwc#347]
Only recently switched to Labwc, and there's a small thing that's been annoying me for a while:
sometimes my mouse clicks are not registering.

I'm mainly talking about mouse clicks in (context) menus.

I think what's happening is that my mouse cursor is moving ever so slightly, so labwc is registering it as a drag event.

**Comment by Consolatis** [comment]
> I think what's happening is that my mouse cursor is moving ever so slightly, so labwc is registering it as a drag event.

That is plausible but drag events should not prevent the actual click to happen.

Tried to verify it by actually starting a drag on a context menu, as in pressing on an item, moving the mouse outside of the menu and then releasing the button. Result: click isn't recognized but I'd argue that is expected behavior.

When the mouse doesn't leave the menu however, it seems to register the click on the item where the cursor was released.

Tested with an X11 application (vlc, menu + context menu on empty playlist) on master and 0.5.2.

**Comment by skerit** [comment]
Interesting. I seem to get different results in Wayland applications.
I've tested it again in Chrome running in XWayland & native Wayland.

On the XWayland Chrome it does seem to work as expected.
Clicking and dragging in a context menu still registers the click on the item where the cursor was released.

On native Wayland Chrome the focus seems to be lost as soon as I start dragging,
the context menu isn't even picking up hovers.

**Comment by Consolatis** [comment]
I wasn't able to reproduce this on latest master. However I \*was\* able to reproduce it on 0.5.2.
Will push a fix for the 0.5 branch shortly. Thanks a lot for making us aware of this issue :)

**Comment by Consolatis** [comment]
PR was merged into the v0.5 branch, would be great if you could give it a test run with something like this:
```
git clone https://github.com/labwc/labwc
cd labwc
git checkout v0.5
meson build
meson compile -C build
```
From a TTY: build/labwc -s foot (or whatever floats your boat instead of foot).

Usually labwc can be tested nested on a runnning labwc (or other compositor) instance, but in this case I assume the issue you discovered with drag operations on wayland native surfaces could also prevent the nested labwc window to work as intended.

If it is still broken for you (or breaks something else) feel free to post here an I'll reopen the issue.
